operators/src/engine/query_processor.rs

```diff
@@ -111,44 +70,32 @@ pub trait PlotQueryProcessor: Sync + Send {
 }
 
 #[async_trait]
-impl<T> QueryProcessor for Box<dyn QueryProcessor<Output = T>> {
-    type Output = T;
-    async fn query<'a>(
-        &'a self,
-        query: QueryRectangle,
-        ctx: &'a dyn QueryContext,
-    ) -> Result<BoxStream<'a, Result<Self::Output>>> {
-        self.as_ref().query(query, ctx).await
-    }
-}
-
-#[async_trait]
-impl<T> QueryProcessor for Box<dyn RasterQueryProcessor<RasterType = T>>
+impl<T> RasterQueryProcessor for Box<dyn RasterQueryProcessor<RasterType = T>>
 where
     T: Pixel,
 {
-    type Output = RasterTile2D<T>;
+    type RasterType = T;
 
-    async fn query<'a>(
+    async fn raster_query<'a>(
```

Comment:
I guess we can still use query then.

Reply:
why do we need it?

Reply:
We still have a common interface for querying stuff.

Reply:
but what is the benefit of using a common query method if both the input and the output types are different?

Reply:
It is not different, we have the same method name, a context and something that is BoxShaped